ShelterModule/Services/Implementations/Announcements/AnnouncementQuery.cs

```diff
@@ -32,7 +32,8 @@ public AnnouncementQuery(PetShareDbContext context)
         if (query.ShelterNames is not null)
             filteredAnnouncements = filteredAnnouncements.Where(a => query.ShelterNames.Contains(a.Pet.Shelter.FullShelterName));
 
-        return await filteredAnnouncements.Select(e => Announcement.FromEntity(e)).ToListAsync(token);
+        var entitiesList = await filteredAnnouncements.Include(x=>x.Author).Include(x=>x.Pet).ToListAsync(token);
```

Comment:
I know we discussed about that, but I don't think it's possible to avoid getting a temporary entitiesList with how I've changed Announcement and Pet objects.